edit/edit_test.go

```diff
@@ -216,3 +217,57 @@ rule()`, 1, true, false, `Error for multiple unnamed rules`},
 		}
 	}
 }
+
+func TestListSubstitute(t *testing.T) {
+	tests := []struct {
+		desc, input, oldPattern, newTemplate, expected string
+	}{
+		{
+			desc:        "no_match",
+			input:       `["abc"]`,
+			oldPattern:  `!!`,
+			newTemplate: `xx`,
+			expected:    `["abc"]`,
+		}, {
+			desc:        "full_match",
+			input:       `["abc"]`,
+			oldPattern:  `.*`,
+			newTemplate: `xx`,
+			expected:    `["xx"]`,
+		}, {
+			desc:        "partial_match",
+			input:       `["abcde"]`,
+			oldPattern:  `bcd`,
+			newTemplate: `xyz`,
+			expected:    `["axyze"]`,
+		}, {
+			desc:        "number_group",
+			input:       `["abcde"]`,
+			oldPattern:  `a(bcd)`,
+			newTemplate: `$1 $1`,
+			expected:    `["bcd bcde"]`,
+		}, {
+			desc:        "name_group",
+			input:       `["abcde"]`,
+			oldPattern:  `a(?P<x>bcd)`,
+			newTemplate: `$x $x`,
+			expected:    `["bcd bcde"]`,
+		},
+	}
+
+	for _, tst := range tests {
+		t.Run(tst.desc, func(t *testing.T) {
+			f, err := build.Parse("BUILD", []byte(tst.input))
+			if err != nil {
+				t.Fatal(err)
+			}
+			lst := f.Stmt[0]
+			oldRegexp := regexp.MustCompile(tst.oldPattern)
```

Comment:
Only use Must* methods as file-level.  Here use Compile and call t.Fatalf if there is an error

Reply:
Done